src/js/preshuffle.s

```diff
@@ -983,8 +987,12 @@ ExitWithoutDrawingEXP:
 .reloc
 RemoveEnemyAndPlaySFX:
   ;; the last attacked enemy is getting cleared out so we want to update the HP display
-  jsr ClearCurrentEnemyHPSlot
-  lda #SFX_MONSTER_HIT
+  ;; Check first if the current slot == this obj slot (stored in $62 at this point)
+  lda CurrentEnemySlot
+  cmp $62
+  bne +
+    jsr ClearCurrentEnemyHPSlot
```

Comment:
Does this potentially add a frame of lag on every enemy kill?  That would add up fast.  What about when first displaying an enemy's name?  Do we have a way to measure these lag frames?